compat/src/memo.js

```diff
@@ -9,7 +9,16 @@ import { shallowDiffers } from './util';
  * @returns {import('./internal').FunctionComponent}
  */
 export function memo(c, comparer) {
-	function shouldUpdate(nextProps) {
+	function Memoed() {}
+
+	Memoed.prototype = new Component();
+	Memoed.displayName = 'Memo(' + (c.displayName || c.name) + ')';
+	// eslint-disable-next-line
+	Memoed.prototype.render = function(props) {
+		return createElement(c, props);
+	};
+
+	Memoed.prototype.shouldComponentUpdate = function(nextProps) {
```

Comment:
Might be worth hoisting this out to a function in module scope so that all `memo()` calls can use the same function reference. Would need to store `comparer` on the instance though I guess, which probably costs too many bytes.

Reply:
Bytes increase is indeed quite on the higher side 😅